### PR TITLE
Fix possible incorrect coordinator info on client

### LINKED ISSUE #19151: Client might pick a light job coordinator it's not connected to
The client picks the coordinator from the member list. But if a member the client isn't connected to is picked, then the message can be actually sent to a different member. In this case the client will have wrong coordinator information and most methods in the `Job` class won't work.

This issue applies easily to non-smart clients, but it also affects normal clients if they temporarily lose connection to some member, or immediately after a new member is added, the client adds the member to member list, but connection is created a bit later.

A reproducer (add to `JobTest`):
```java
    @Test
    public void test_nonSmartClient() {
        HazelcastInstance client = factory().newHazelcastClient(configForNonSmartClientConnectingTo(instance()));

        // try multiple times - we test the case when the client randomly picks a member it's not connected to.
        // It should not pick such a member.
        for (int i = 0; i < 10; i++) {
            Job job = client.getJet().newLightJob(streamingDag());
            assertTrueEventually(() -> assertJobExecuting(job, instance()));
            cancelAndJoin(job);
        }
    }
```

### PR DESCRIPTION
**Pull request by viliam-durina**
The fix is twofold:

- choose the coordinator UUID among active connections, not among active
members

- include the UUID in the SubmitJob message, the member receiving the
message will then forward the request to the correct member. This
handles the case when the connection was lost in the meantime since we
chose that connection and sent the message.

Fixes #19151
Client Protocol PR: https://github.com/hazelcast/hazelcast-client-protocol/pull/387/

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java [hunk 1]
```diff
@@ -98,6 +99,7 @@
         this.container = container;
         this.lightJobCoordinator = isLightJob ? findLightJobCoordinator() : null;
         this.logger = loggingService().getLogger(Job.class);
+        logger.info("aaa coordinator for " + idToString(jobId) + " is " + lightJobCoordinator);
```
Comment (olukas) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java:102]: probably leftover

**Review by olukas — APPROVED** [review]
Thanks. Related issue seems to be fixed by this PR.

File: hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java (hunk 1 above)
Comment (viliam-durina) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java]: That was a test if you're paying attention - you passed 😉 
Removed.

**Review by ufukyilmaz — APPROVED** [review]
LGTM

**Comment by viliam-durina** [comment]
run-lab-run

**Review by sancar — APPROVED** [review]

**Pull request merged**